test/e2e/framework/job/wait.go

```diff
@@ -68,6 +72,18 @@ func WaitForJobComplete(ctx context.Context, c clientset.Interface, ns, jobName
 	})
 }
 
+// WaitForJobSuspend uses c to wait for suspend condition for the Job jobName in namespace ns.
+func WaitForJobSuspend(ctx context.Context, c clientset.Interface, ns, jobName string) error {
+	return WaitForJobCondition(ctx, c, ns, jobName, "suspended", JobTimeout, func(job *batchv1.Job) (bool, error) {
```

Comment:
```suggestion
	return WaitForJobState(ctx, c, ns, jobName, "suspended", JobTimeout, func(job *batchv1.Job) (bool, error) {
```

?

`Condition` confused me a little, just because there is the `job.Status.Conditions` field. But IIUC, this is supposed to be more generic.

Reply:
updated.